### DISCUSSION TITLE
Rows getting randomly ordered

### DISCUSSION THREAD

**Discussion by d41ly** [nocodb/nocodb#6000] [Q&A]
Hey guys, pretty confused by this 'feature'. Say, I've uploaded a CSV dataset into NocoDB, it has a particular order of rows (obviously). However, on reload, without using any sorting features (which, by the way, just don't work as far as I understand?) or whatever, the rows would just get reordered by whatever logic. Could someone help me understand why this happens and if anything can be done to rectify this behavior?

**Comment by dstala** [comment]
Can you help us with a redacted sample CSV may be? and also your project configuration settings - you can copy this from project menu `Copy Project Info` option

**Comment by d41ly** [comment]
Node: **v16.20.0**
Arch: **x64**
Platform: **linux**
Docker: **true**
RootDB: **pg**
PackageVersion: **0.109.3**

@dstala sharing the project info above. I'd have to heavily redact the csv, thus I'm not sure how useful it'd actually be, however I'm sharing a screenshot of the column types, perhaps that could be of some use. So far, any table I've tested had this problem. Even if I would rearrange the rows by hand, as in, manually delete/create/populate (can we have a row drag to move feature? how cool would it be :)), they'd STILL get rearranged on future refresh, thus it doesn't look like the problem is import related.
![image](https://github.com/nocodb/nocodb/assets/138282356/5197ea0d-d976-46d0-b888-2c6c1d8d29a1)

**Comment by d41ly** [comment]
After playing around with the tables for a bit, I came to a belief that there is some kind of sorting that exists by default, but I'm can't figure a way of how to prevent it from occuring. If I add a row somewhere, or upload a new CSV, and then refresh the app, the rows are rearranged. I rolled back to an earlier build 0.109.1 where the problem persists, but sorting actually works (doesn't work for me in 0.109.3), and applied some sorting parameters — the random rearranging stopped and the tables displayed in the way they were sorted. 
In my opinion, there shouldn't be any kind of sorting enabled by default and tables should retain their original row order, which they should return to when the sorting paramerets are removed.
